spec/timezone.html

```diff
@@ -213,9 +202,28 @@ <h1>Temporal.TimeZone.prototype.getInstantFor ( _dateTime_ [ , _options_ ] )</h1
             1. Return _possibleInstants_[0].
           1. If _disambiguation_ is *"later"*, then
             1. Return _possibleInstants_[_n_ − 1].
-          1. If _disambiguation_ is *"reject"*, then
-            1. Throw a *RangeError* exception.
-        1. <mark>TODO - handle n=0 (skipped times like Spring DST).</mark>
+          1. Assert: _disambiguation_ is *"reject"*.
+          1. Throw a *RangeError* exception.
+        1. Assert: _n_ = 0.
+        1. If _disambiguation_ is *"reject"*, then
+          1. Throw a *RangeError* exception.
+        1. Let _epochNanoseconds_ be ! GetEpochFromParts(_dateTime_.[[ISOYear]], _dateTime_.[[ISOMonth]], _dateTime_.[[ISODay]], _dateTime_.[[ISOHour]], _dateTime_.[[ISOMinute]], _dateTime_.[[ISOSecond]], _dateTime_.[[ISOMillisecond]], _dateTime_.[[ISOMicrosecond]], _dateTime_.[[ISONanosecond]]).
+        1. Let _dayBefore_ be ? CreateTemporalInstant(_epochNanoseconds_ − 8.64 × 10<sup>13</sup>).
+        1. Let _dayAfter_ be ? CreateTemporalInstant(_epochNanoseconds_ + 8.64 × 10<sup>13</sup>).
+        1. Let _offsetBefore_ be ? GetOffsetNanosecondsFor(_timeZone_, _dayBefore_).
+        1. Let _offsetAfter_ be ? GetOffsetNanosecondsFor(_timeZone, _dayAfter_).
+        1. Let _nanoseconds_ be _offsetAfter_ − _offsetBefore_.
+        1. If _disambiguation_ is *"earlier"*, then
+          1. Let _earlier_ be ? AddDateTime(_dateTime_.[[ISOYear]], _dateTime_.[[ISOMonth]], _dateTime_.[[ISODay]], _dateTime_.[[ISOHour]], _dateTime_.[[ISOMinute]], _dateTime_.[[ISOSecond]], _dateTime_.[[ISOMillisecond]], _dateTime_.[[ISOMicrosecond]], _dateTime_.[[ISONanosecond]], 0, 0, 0, 0, 0, 0, 0, 0, 0, −_nanoseconds_, *"constrain"*).
+          1. Set _possibleInstants_ to ? GetPossibleInstantsFor(_timeZone_, _earlier_).
+          1. Assert: _possibleInstants_'s length is not 0.
+          1. Return _possibleInstants_[0].
+        1. Assert: _disambiguation_ is *"compatible"* or *"later"*.
+        1. Let _later_ be ? AddDateTime(_dateTime_.[[ISOYear]], _dateTime_.[[ISOMonth]], _dateTime_.[[ISODay]], _dateTime_.[[ISOHour]], _dateTime_.[[ISOMinute]], _dateTime_.[[ISOSecond]], _dateTime_.[[ISOMillisecond]], _dateTime_.[[ISOMicrosecond]], _dateTime_.[[ISONanosecond]], 0, 0, 0, 0, 0, 0, 0, 0, 0, _nanoseconds_, *"constrain"*).
+        1. Set _possibleInstants_ to ? GetPossibleInstantsFor(_timeZone_, _later_).
+        1. Set _n_ to _possibleInstants_'s length.
+        1. Assert: _n_ is not 0.
```

Comment:
You can't know that - `GetPossibleInstantsFor` can return whatever it wants.

Reply:
Hmm, you're right. I wonder what is best to do in this case. If getPossibleInstantsFor is inconsistent, then I think probably the best thing to do is throw.

Reply:
I got rid of the assertions, in any case.